### PR TITLE
Add build check on PR

### LINKED ISSUE #10: Add build check on PR
Verify that `./repo.sh` docs finish without error

### PR DESCRIPTION
**Pull request by rollandf**
Fix #10

### CODE REVIEW

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  build:
+    name: generate-html
+    runs-on: ubuntu-22.04
+    permissions:
+      contents: read
+    steps:
+      - name: checkout
+        uses: actions/checkout@v4
+      - name: build
+        run: ./repo.sh docs
```
Comment (ykulazhenkov) [inline .github/workflows/build.yaml:14]: Maybe we can improve this job to store generated docs as a build artifact?

Comment (rollandf) [reply .github/workflows/build.yaml:14]: Sure, any suggestion?
Should it be in same PR or follow up?

Comment (ykulazhenkov) [reply .github/workflows/build.yaml:14]: here is the example how to achieve this with gh actions https://docs.github.com/en/actions/using-workflows/storing-workflow-data-as-artifacts#example.
Not a blocker for to this PR

**Comment by rollandf** [comment]
> How can I make this fail?

For example:

```
[ERROR] Documentation build failed.
Errors (from: '/home/frolland/git/network-operator-docs/_build/docs/sphinx_warnings.txt'):
/home/frolland/git/network-operator-docs/docs/device-plugin.rst:20: WARNING: Title overline too short.

**********
Device Plugin
**********

```

File: .github/workflows/build.yaml (hunk 1 above)
Comment (rollandf) [reply .github/workflows/build.yaml:14]: Let's handle it as part of #5

**Review by killianmuldoon — APPROVED** [review]

**Pull request merged**